pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -191,7 +192,14 @@ public CompletableFuture<Boolean> canConsumeAsync(TopicName topicName, String ro
             if (isSuperUser) {
                 return CompletableFuture.completedFuture(true);
             } else {
-                return provider.canConsumeAsync(topicName, role, authenticationData, subscription);
+                AuthenticationDataSource subscriptionDataSource;
+                if (authenticationData instanceof AuthenticationDataSubscription) {
+                    subscriptionDataSource = authenticationData;
+                } else {
+                    subscriptionDataSource = new AuthenticationDataSubscription(authenticationData, subscription);
+                }
+                return provider.allowTopicOperationAsync(topicName, role, TopicOperation.CONSUME,
+                        subscriptionDataSource);
```

Comment:
Can we deprecate this `canConsumeAsync` method? It seems like the same kind of logic that we're deprecating in the `AuthorizationProvider`.